### DISCUSSION TITLE
Login with Fetch in Javascript

### DISCUSSION THREAD

**Discussion by MichaelHarmsTIB** [ckan/ckan#7130] [Q&A]
Hello,
I create a login with Postman, which works fine. If I try it over fetch with Javascript, I got no successful login.
  var myHeaders = new Headers();
  myHeaders.append('Cookie','auth_tkt="fdac0b9a144b596245a30fdf6b3bc8fe63371f8cckan!"; ckan=84b54f303e7820c8d8f6c7adae9c96d7bb46708b03af417de2e34c42be4756229df14fa4');
  myHeaders.append('Content-Type', 'application/x-www-form-urlencoded');
  myHeaders.append('Accept', '*/*');
  myHeaders.append('Connection', 'keep-alive');

  const login = () => {
    fetch('http://192.168.59.5/login_generic', {
      method: 'POST',
      credentials: "include",
      redirect: 'follow',
      headers: myHeaders,
      body: JSON.stringify({
        'came_from': '/user/logged_in',
       'login': 'ckan',
       'password': '12345678',
       'remember': '63072000' 
       })
    })
      .then(res => {
        console.log(res);
      })
      .catch(err => console.error(err))
Has anyone an idea why it doesn't works with fetch?
Regards
M

**Comment by smotornyuk** [comment]
Are you sure that you are requesting a login endpoint from the host, explicitly allowed via the `Access-control-allow-origin` header set by your CKAN app? It's the first thing that comes to mind, when I see `credentials: "include"`. 
BTW, why do you think that it doesn't work with fetch? I see no error mentioned in your report, which means that `.catch(err => console.error(err))` line was not triggered and there are no errors in your request.
